membership-attribute-service/app/services/PaymentFailureAlerter.scala

```diff
@@ -53,13 +54,14 @@ object PaymentFailureAlerter extends SafeLogging {
         case None => Future.successful(None)
       }
 
-      def getProductDescription(subscription: Subscription[SubscriptionPlan.AnyPlan]) = if (subscription.asMembership.isDefined) {
```

Comment:
`asMembership` is deleted elsewhere in this PR, but in the end it was just checking the product and downcasting in the vein of a pattern match, so now I've just made it do the check and not worry about the type system